packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,363 @@
+///  * [OpenUpwardsPageTransitionsBuilder], which defines a page transition
+///    that's similar to the one provided by Android P.
+///  * [CupertinoPageTransitionsBuilder], which defines a horizontal page
+///    transition that matches native iOS page transitions.
+@immutable
+class PageTransitionsTheme extends Diagnosticable {
+  /// Construct a PageTransitionsTheme.
+  ///
+  /// By default the list of builders is: [FadeUpwardsPageTransitionsBuilder],
+  /// [CupertinoPageTransitionsBuilder] for [TargetPlatform.android]
+  /// and [TargetPlatform.iOS] respectively.
+  const PageTransitionsTheme({ Map<TargetPlatform, PageTransitionsBuilder> builders }) : _builders = builders;
+
+  static final Map<TargetPlatform, PageTransitionsBuilder> _defaultBuilders = <TargetPlatform, PageTransitionsBuilder>{
+    TargetPlatform.android: const FadeUpwardsPageTransitionsBuilder(),
+    TargetPlatform.iOS: const CupertinoPageTransitionsBuilder(),
+  };
+
+  /// The [PageTransitionsBuilder]s supported by this theme.
+  Map<TargetPlatform, PageTransitionsBuilder> get builders => _builders ?? _defaultBuilders;
+  final Map<TargetPlatform, PageTransitionsBuilder> _builders;
+
+  /// Delegates to the builder for the current [ThemeData.platform]
+  /// or [FadeUpwardsPageTransitionsBuilder].
+  ///
+  /// [MaterialPageRoute.buildTransitions] delegates to this method.
+  Widget buildTransitions<T>(
+    PageRoute<T> route,
+    BuildContext context,
+    Animation<double> animation,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    TargetPlatform platform = Theme.of(context).platform;
+
+    if (CupertinoPageRoute.isPopGestureInProgress(route))
+      platform = TargetPlatform.iOS;
+
+    final PageTransitionsBuilder matchingBuilder =
+      builders[platform] ?? const FadeUpwardsPageTransitionsBuilder();
+    return matchingBuilder.buildTransitions<T>(route, context, animation, secondaryAnimation, child);
+  }
+
+  List<PageTransitionsBuilder> _all(Map<TargetPlatform, PageTransitionsBuilder> builders) {
+    return TargetPlatform.values.map((TargetPlatform platform) => builders[platform]).toList();
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final PageTransitionsTheme typedOther = other;
+    if (identical(builders, other.builders))
+      return true;
+    return listEquals<PageTransitionsBuilder>(_all(builders), _all(typedOther.builders));
+  }
+
+  @override
+  int get hashCode => hashList(_all(builders));
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    const PageTransitionsTheme defaultTheme = PageTransitionsTheme();
+    properties.add(DiagnosticsProperty<Map<TargetPlatform, PageTransitionsBuilder>>('builders', builders, defaultValue: defaultTheme.builders));
```

Comment:
you could just hand in _defaultBuilders directly instead of the dance here with `defaultTheme`, no?

Reply:
Yes, done.